src/peripheral.ts

```diff
@@ -47,11 +55,23 @@ export class Peripheral implements Ble {
     return onReceivedNotificationListener
   }
 
-  async advertise() {
-    try {
-      await sdk.advertise({})
-    } catch (e) {
-      throw new Error('An error occurred while trying to advertise: ' + e)
-    }
+  public registerOnConnectedListener(
+    cb: ({ identifier, name }: { identifier: string; name?: string }) => void
+  ) {
+    const onConnectedPeripheralListener = this.bleDidcommEmitter.addListener(
+      'onConnectedCentral',
+      cb
+    )
+    return onConnectedPeripheralListener
+  }
+
+  public registerOnDisconnectedListener(
```

Comment:
More a general question about this. Maybe it would make sense to merge onConnected and onDisconnected into a single listener? If that is even possible? The there'd be a listener for the connection and instead of having those separate we could just pass a callback for onConnect and a different one for onDisconnect. But might be missing something/resons why it is this way.

Reply:
Yeah that's possible but I miss the point as to why.

What's the benefit to:

```typescript
const registerConnectedAndDisconnectedListener(conn: () => void, disconn: () => void) {
  ...
}
```
vs
```typescript
const registerConnected(cb: () => void) {
  ...
}
const registerDisonnected(cb: () => void) {
  ...
}

Reply:
Just thought it's cleaner to use